lib/trino-parquet/src/test/java/io/trino/parquet/reader/TestEncryption.java

```diff
@@ -0,0 +1,345 @@
+        decryptAndValidate();
+    }
+
+    @Test
+    public void testOneRecord()
+            throws IOException
+    {
+        MessageType schema = createSchema();
+        String[] encryptColumns = {"name", "gender"};
+        this.inputFile = new EncryptionTestFileBuilder(conf, schema)
+                .withEncryptColumns(encryptColumns)
+                .withNumRecord(1)
+                .withCodec("GZIP")
+                .withPageSize(1000)
+                .withFooterEncryption()
+                .build();
+        decryptAndValidate();
+    }
+
+    @Test
+    public void testMillionRows()
+            throws IOException
+    {
+        MessageType schema = createSchema();
+        String[] encryptColumns = {"name", "gender"};
+        this.inputFile = new EncryptionTestFileBuilder(conf, schema)
+                .withEncryptColumns(encryptColumns)
+                .withNumRecord(1000000)
+                .withCodec("GZIP")
+                .withPageSize(1000)
+                .withFooterEncryption()
+                .build();
+        decryptAndValidate();
+    }
+
+    @Test
+    public void testPlainTextFooter()
+            throws IOException
+    {
+        MessageType schema = createSchema();
+        String[] encryptColumns = {"name", "gender"};
+        this.inputFile = new EncryptionTestFileBuilder(conf, schema)
+                .withEncryptColumns(encryptColumns)
+                .withNumRecord(10000)
+                .withCodec("ZSTD")
+                .withPageSize(1000)
+                .build();
+        decryptAndValidate();
+    }
+
+    @Test
+    public void testLargePageSize()
+            throws IOException
+    {
+        MessageType schema = createSchema();
+        String[] encryptColumns = {"name", "gender"};
+        this.inputFile = new EncryptionTestFileBuilder(conf, schema)
+                .withEncryptColumns(encryptColumns)
+                .withNumRecord(100000)
+                .withCodec("GZIP")
+                .withPageSize(100000)
+                .withFooterEncryption()
+                .build();
+        decryptAndValidate();
+    }
+
+    @Test
+    public void testAesGcmCtr()
+            throws IOException
+    {
+        MessageType schema = createSchema();
+        String[] encryptColumns = {"name", "gender"};
+        this.inputFile = new EncryptionTestFileBuilder(conf, schema)
+                .withEncryptColumns(encryptColumns)
+                .withNumRecord(100000)
+                .withCodec("GZIP")
+                .withPageSize(1000)
+                .withEncrytionAlgorithm(ParquetCipher.AES_GCM_CTR_V1)
+                .build();
+        decryptAndValidate();
+    }
+
+    private MessageType createSchema()
+    {
+        return new MessageType("schema",
+                new PrimitiveType(OPTIONAL, INT64, "id"),
+                new PrimitiveType(REQUIRED, BINARY, "name"),
+                new PrimitiveType(OPTIONAL, BINARY, "gender"));
+    }
+
+    private void decryptAndValidate()
```

Comment:
After we update EncryptionTestFileBuilder to write actual files, this should be updated to read from files using code similar to `TestTimestamp#testReadingAs`